pkgs/development/libraries/glog/default.nix

```diff
@@ -33,12 +24,21 @@ stdenv.mkDerivation rec {
   cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" ];
 
   checkInputs = [ perl ];
-  doCheck = false; # fails with "Mangled symbols (28 out of 380) found in demangle.dm"
+  doCheck = true;
+  # Add `build/` directory (`$PWD`) to LD_LIBRARY_PATH so that the built library
+  # can be loaded for the tests.
+  # Adding to LD_LIBRARY_PATH correctly is nontrivial:
+  #   https://jdhao.github.io/2021/07/03/ld_library_path_empty_item/
+  #   https://unix.stackexchange.com/questions/162891/append-to-path-like-variable-without-creating-leading-colon-if-unset
+  preCheck = ''
+    export LD_LIBRARY_PATH=''${LD_LIBRARY_PATH:+''${LD_LIBRARY_PATH}:}''$PWD
```

Comment:
I think an easier way to do this is to add `-DCMAKE_SKIP_BUILD_RPATH=OFF` to cmakeFlags. There is a treewide PR for this with some more info: https://github.com/NixOS/nixpkgs/pull/108496

Reply:
Done. I've added it as a commit on top so it can easily be reverted if needed.